src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer.StateManager.cs

```diff
@@ -20,6 +21,12 @@ internal partial class DiagnosticIncrementalAnalyzer
         /// </summary>
         private partial class StateManager
         {
+            // this cache is supposed to be most of time empty except when we are doing build-live deduplication.
+            // since build is not something that happens continuously, we do not want to hold this onto memory all the time. we will build it once
+            // per build even if it means we are building same one as last time. this cache will go away as soon as solution snapshot moves which
+            // is quite frequent since it includes thing like open/close document, typing, file change and etc.
+            private static readonly ConditionalWeakTable<Project, List<StateSet>> s_buildOnlyProjectStateSetCache = new ConditionalWeakTable<Project, List<StateSet>>();
```

Comment:
We might need to do performance measurements - last time I used CWT from Project to CompilationWithAnalyzers, it caused VM to rise quite a bit.